base/namedtuple.jl

```diff
@@ -404,11 +407,35 @@ end
     types = diff_types(a, names)
     n = length(names)
     A = Vector{Any}(undef, n)
-    for i=1:n
+    for i = 1:n
         n = names[i]
         A[i] = getfield(a, n)
     end
-    _new_NamedTuple(NamedTuple{names, types}, (A...,))
+    _new_NamedTuple(NamedTuple{names,types}, (A...,))
+end
+
+"""
+    delete(a::NamedTuple, keys::Union{Symbol, Tuple{Vararg{Symbol}}})
+
+Construct a new named tuple from `a` by removing the named entries.
+
+```jldoctest
+julia> delete((a=1, b=2, c=3), :a)
+(b = 2, c = 3)
+julia> delete((a=1, b=2, c=3), :b)
+(a = 1, c = 3)
+julia> delete((a=1, b=2, c=3), (:c, :b))
+(a = 1,)
+```
+"""
+function delete(a::NamedTuple{an}, @nospecialize(key::Symbol)) where {an}
```

Comment:
Why is there no specialize on `key::Symbol`?

Reply:
Thank you. I am removing it.